vulnerability/replicator.go

```diff
@@ -0,0 +1,71 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package vulnerability
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	"github.com/elastic/cloudbeat/resources/providers/awslib/ec2"
+	"github.com/elastic/elastic-agent-libs/logp"
+)
+
+type VulnerabilityReplicator struct {
+	log      *logp.Logger
+	provider *ec2.Provider
```

Comment:
This was done in order to code share as much as I can with the pre-existing code.

Reply:
The code sharing is actually good, the implementation should remain the same under `ec2.Provider` all I'm saying is that instead of holding a pointer `ec2.Provider` you should hold an interface `SnapshotProvider` (or your suggestion) and the underlying type will be `ec2.Provider`.